addons/better-terrain/BetterTerrain.gd

```diff
@@ -600,6 +625,31 @@ func get_tiles_in_terrain(ts: TileSet, type: int) -> Array[TileData]:
 	return result
 
 
+## Returns an [Array] of [Dictionary] items including information about each 
+## tile included in the specified terrain [code]type[/code] for 
+## the [TileSet] [code]ts[/code]. Each Dictionary item includes 
+## [TileSetAtlasSource] [code]source[/code], [TileData] [code]td[/code], 
+## [Vector2i] [code]coord[/code], and [int] [code]alt_id[/code].
+func get_tile_sources_in_terrain(ts: TileSet, type: int) -> Array[Dictionary]:
+	var result:Array[Dictionary] = []
+	
+	var cache := _get_cache(ts)
+	var tiles = cache[type]
+	if !tiles:
+		return result
+	for c in tiles:
+		var source := ts.get_source(c[0]) as TileSetAtlasSource
```

Comment:
Technically this can return null, if the user deletes tile atlases prior to this function getting run. I think there's a couple places where it isn't null checked at the moment (my bad, I should fix that). Just needs to continue if it's null.